operator/v1alpha1/types_kueue.go

```diff
@@ -0,0 +1,243 @@
+// +genclient:nonNamespaced
+// +kubebuilder:storageversion
+// +kubebuilder:subresource:status
+// +kubebuilder:validation:XValidation:rule="self.metadata.name == 'cluster'",message="olm is a singleton, .metadata.name must be 'cluster'"
+type Kueue struct {
+	metav1.TypeMeta `json:",inline"`
+	// metadata for kueue
+	// +optional
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// spec holds user settable values for configuration
+	// kueue configuration must not be changed once the object exists
+	// to change the configuration, one can delete the object and create a new object.
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="values are immutable once set"
+	// +required
+	Spec KueueOperandSpec `json:"spec"`
+	// status holds observed values from the cluster. They may not be overridden.
+	// +optional
+	Status KueueStatus `json:"status,omitempty"`
+}
+
+type KueueOperandSpec struct {
+	operatorv1.OperatorSpec `json:",inline"`
+	// config is the desired configuration
+	// for the kueue operator.
+	// +required
+	Config KueueConfiguration `json:"config"`
+}
+
+// +kubebuilder:validation:Enum=QueueNameOptional;QueueNameRequired
+type ManageJobsWithoutQueueNameOption string
+
+// ManageJobsWithoutQueueName allows to control what kind of
+// jobs kueue will manage.
+// Kueue jobs usually require kueue.x-k8s.io/queue-name on each job
+// to be opt-in for Kueue.
+const (
+	// QueueNameOptional means kueue will assume all workloads
+	// are to be gated by Kueue.
+	// This must be used with ManagedJobsNamespaceSelector.
+	QueueNameOptional ManageJobsWithoutQueueNameOption = "QueueNameOptional"
+	// QueueNameRequired means that the jobs require a queue label.
+	QueueNameRequired ManageJobsWithoutQueueNameOption = "QueueNameRequired"
+)
+
+// +kubebuilder:validation:Enum=Enabled;Disabled
+type EnabledOrDisabled string
+
+const (
+	Enabled  EnabledOrDisabled = "Enabled"
+	Disabled EnabledOrDisabled = "Disabled"
+)
+
+type KueueConfiguration struct {
+	// integrations are the types of integrations Kueue will manage
+	// +required
+	Integrations Integrations `json:"integrations"`
+	// resources provides additional configuration options for handling the resources.
+	// Supports https://github.com/kubernetes-sigs/kueue/blob/release-0.10/keps/2937-resource-transformer/README.md
+	// +optional
+	Resources Resources `json:"resources,omitempty"`
+	// manageJobsWithoutQueueName controls whether or not Kueue reconciles
+	// jobs that don't set the annotation kueue.x-k8s.io/queue-name.
+	// +kubebuilder:default=QueueNameRequired
+	// +optional
+	ManageJobsWithoutQueueName *ManageJobsWithoutQueueNameOption `json:"manageJobsWithoutQueueName,omitempty"`
+	// managedJobsNamespaceSelector can be used to omit some namespaces from ManagedJobsWithoutQueueName
+	// Only valid if ManagedJobsWithoutQueueName is QueueNameOptional
+	// +optional
+	ManagedJobsNamespaceSelector *metav1.LabelSelector `json:"managedJobsNamespaceSelector,omitempty"`
```

Comment:
Still interested in thoughts on this being a nested option as both these fields are related to job management configurations

Reply:
I prefer to not differ much from upstream. If we feel that this API is not useful we can propose a new one in upstream but I'd like to not have to do too much conversions especially since this is a beta configuration API.

Reply:
Their API is beta and subject to change, our API is not going to go beta right? It's going to v1alpha1 or v1?

If we ship a v1 API, its shape can never change in the future, this is a risk of building on something that isn't v1.

I think we need to build a good abstraction that works for us, and convert it for perpetuity, else upstream changes to the API will leave us stale and we will still need conversion

Reply:
Dropping these fields for now.